tests/security/pam/pam_mount.pm

```diff
@@ -0,0 +1,130 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: PAM tests for pam-mount, the encrypted volume should be mounted
+#          and unmounted during user login and logout
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#70345, tc#1767581
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+use utils 'script_run_interactive';
+use base 'consoletest';
+
+sub run {
+    # There is some issue with script_run_interactive script,
+    # it fails to match the serial console output sometimes,
+    # so switch to root-console here
+    select_console 'root-console';
+
+    # Install the pam-mount package, since this service package is not installed by default
+    zypper_call 'in pam_mount';
+
+    # Define the uesr and encrypt key for the volume
+    my $user     = 'bernhard';
+    my $key      = 'SUSE_t595_qw';
+    my $testfile = 'testfile';
+    my $loopdev  = '/dev/loop0';
```

Comment:
a suggestion: I think "losetup -f" is better than "/dev/loop0", if not, maybe 'assert_script_run "losetup $loopdev $testfile";' aborts unexpectedly.

Reply:
Yup, nice comment, will modify it

Reply:
my $loopdev  = script_output 'losetup -f';